crates/ra_ide/src/references.rs

```diff
@@ -122,8 +123,12 @@ pub(crate) fn find_all_refs(
     let mut search_kind = ReferenceKind::Other;
 
     if let TokenAtOffset::Between(ref left, ref right) = token {
-        if (right.kind() == SyntaxKind::L_CURLY || right.kind() == SyntaxKind::L_PAREN)
-            && left.kind() != SyntaxKind::IDENT
+        if (((right.kind() == SyntaxKind::WHITESPACE
+            && right.next_token().map(|t| t.kind()) == Some(SyntaxKind::L_CURLY))
+            || right.kind() == SyntaxKind::L_CURLY)
```

Comment:
What happens if we have two white space tokens here ?

Reply:
There should never be more than one `WHITESPACE` token in a row, so this seems reasonable (although the if condition is a bit large).